src/main/java/game/entities/items/weapons/Damage.java

```diff
@@ -1,15 +1,28 @@
 package game.entities.items.weapons;
 
 import game.entities.Entity;
+import game.entities.items.Item;
 
 public class Damage {
     private Entity target;
+    private Weapon weapon;
 
     public Damage(Entity target) {
         this.target = target;
     }
 
-    public void dealDamage(int damageDealt) { target.setHP(target.getHP() - damageDealt); }
+    public Damage(Weapon weapon) {
+        this.weapon = weapon;
+    }
+
+    public Damage addDamage(int damageDealt) { 
+        target.setHP(target.getHP() - damageDealt); 
```

Comment:
Fix this so that `target` must be set in the constructor because if the `addDamage` gets called with `target` being `null`, it will throw an error.

Reply:
It's been there asshole